### PR TITLE
Progress bar feature

### PR DESCRIPTION
**Pull request by Maofryy**
# New Feature
Changes proposed in this pull request:
 ```python
info_progress_bar(prefix: str, value: float, max_value: float)
```
Adding to ```info_progress``` a progress bar with ascii character "█" between prefix and percent

### CODE REVIEW

File: cli_ui/__init__.py [hunk 1]
```diff
@@ -378,6 +378,25 @@ def info_progress(prefix: str, value: float, max_value: float) -> None:
...
+    :param max_value: the max value
+    :param prefix: the prefix message to print
+
+
+    """
+    if sys.stdout.isatty():
+        sys.stdout.write("{prefix}: |{done}{spaces}| {percent:>3.0f}%\r".format(
+            prefix=prefix,
+            done = int(value*2)*"█",
+            spaces = int(-1 * (max_value*2 - int(value*2)) // 1 * -1)*" ",
```
Comment (dmerejkowsky) [inline cli_ui/__init__.py]: This would be easier to read with the introduction of a few intermediary variables and a few comments

**Comment by dmerejkowsky** [comment]
Looks like a good addition, but there are a few problems:

* Tests are missing
* The code does not seem to work on my machine

screenshot

* I fear it would not work on Windows (I need to check)
* Finally, there's already a pretty good library for that called `tqdm` which complements `cli_ui` nicely I think

**Review by dmerejkowsky — CHANGES_REQUESTED** [review]

**Comment by Maofryy** [comment]
- I implemented tests and needed to change the function to do so. Its passing fine, still tests fails on windows not recognizing colored text.
-  Previously I had a dynamic size depending on the ```max_value``` but now i fixed it so the bar has only ```MAX_SIZE``` characters
- I tested it manually on ```cmd.exe``` and it is working.
- I was inspired by the ```tqdm.tqdm``` module yes, but I want to mix the two librairies by displaying a progress bar with different colors according to progress (dark red at 0% into bright green at 100%) but I still need to think about its implementation and tests (here on my windows the colored tests are failing).

**Comment by Maofryy** [comment]
* Merged #65 with the current pr
* Implemented rather bluntly a red-green gradient to the progress bar, scaling with the ```MAX_SIZE``` (maybe add the max_size as an argument ?)
* Updated test to work with color

**Comment by dmerejkowsky** [comment]
Hello again.

I've just merged a bunch of stuff in the main branch - so you should incorporate them in your PR.

You can do it by rebasing your branch on top of `origin/master` and force-push this branch (this will be easier to review that way).

You ll see that the linters have caught a few additional errors, like this:

```
cli_ui/__init__.py:357:51: N803 argument name 'fileObj' should be lowercase
``` 

so you'll have to fix them too.

Also, you should add a call to `progress_bar` in `main_demo`, and add a section about it in the documentation.

Finally, you should update the changelog.

**Comment by dmerejkowsky** [comment]
Oh - and also found a crash. It happens if `current_value`  is bigger than `max_value` 

```
  File "/mnt/data/dmerej/src/python-cli-ui/cli_ui/__init__.py", line 712, in main_demo
    info_progress_bar("Progress", 1000, 100)
  File "/mnt/data/dmerej/src/python-cli-ui/cli_ui/__init__.py", line 383, in info_progress_bar
    color = COLOR_LIST[int(percent / 12.5)]
IndexError: list index out of range
```

File: cli_ui/__init__.py [hunk 2]
```diff
@@ -355,6 +355,47 @@ def info_progress(prefix: str, value: float, max_value: float) -> None:
...
+        colorama.Fore.GREEN + colorama.Style.DIM,
+        colorama.Fore.GREEN + colorama.Style.NORMAL,
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+    ]
+    COLOR_RESET = colorama.Fore.WHITE + colorama.Style.NORMAL
+
+    percent = float(value) / max_value * 100
+    color = COLOR_LIST[int(percent / 12.5)]
```
Comment (dmerejkowsky) [inline cli_ui/__init__.py:385]: * This won't work if `percent` is bigger than 100
* I'm not sure why you go through a percentage anyway. You can just use `ratio =value / max_value` instead I think.
* You don't need to hard-code 12.5 - use `len(COLOR_LIST)` instead

File: cli_ui/__init__.py [hunk 3]
```diff
@@ -355,6 +355,47 @@ def info_progress(prefix: str, value: float, max_value: float) -> None:
...
+) -> None:
+    """ Display info progress as a bar.
+
+    :param value: the current value
+    :param max_value: the max value
+    :param prefix: the prefix message to print
+
+
+    """
+    MAX_SIZE = 10
```
Comment (dmerejkowsky) [inline cli_ui/__init__.py:369]: rename to `bar_length` ?

File: cli_ui/__init__.py [hunk 4]
```diff
@@ -355,6 +355,47 @@ def info_progress(prefix: str, value: float, max_value: float) -> None:
...
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+        colorama.Fore.GREEN + colorama.Style.BRIGHT,
+    ]
+    COLOR_RESET = colorama.Fore.WHITE + colorama.Style.NORMAL
+
+    percent = float(value) / max_value * 100
+    color = COLOR_LIST[int(percent / 12.5)]
+    value_size = int((float(value) * MAX_SIZE) / max_value)
+    if fileObj.isatty():
```
Comment (dmerejkowsky) [inline cli_ui/__init__.py:387]: I think you can use the `colors_enabled`  and the `write_and_flush`  helper methods here

**Review by dmerejkowsky — CHANGES_REQUESTED** [review]

**Pull request closed without merging**